posthog/models/messaging.py

```diff
@@ -0,0 +1,29 @@
+import hashlib
+
+from django.conf import settings
+from django.db import models
+
+from .utils import UUIDModel
+
+
+class MessagingRecordManager(models.Manager):
+    def get_or_create(self, defaults=None, **kwargs):
+        raw_email = kwargs.pop("raw_email", None)
+
+        if raw_email:
+            kwargs["email_hash"] = hashlib.sha256(f"{settings.SECRET_KEY}_{raw_email}".encode()).hexdigest()
+
+        return super().get_or_create(defaults, **kwargs)
+
+
+class MessagingRecord(UUIDModel):
+
+    objects = MessagingRecordManager()
+
+    email_hash: models.CharField = models.CharField(max_length=1024)
+    campaign_key: models.CharField = models.CharField(max_length=128)
+    sent_at: models.DateTimeField = models.DateTimeField(null=True)
+    created_at: models.DateTimeField = models.DateTimeField(auto_now_add=True)
+
+    class Meta:
+        unique_together = ("email_hash", "campaign_key")  # can only send campaign once to each email
```

Comment:
This looks a lot `UserMessagingRecord` in `posthog-production`, but email address- instead of user-centric. Could we maybe do user-centric here as well to use the keep track of messaging using a single scheme across both repos?

Reply:
I've already created an issue (#1983) to remind us to remove `UserMessagingRecord` in favor of this model. As for why no user-centric, I do believe we consider this improvement but it has trade-offs, the reason for doing this email-centric at first was due to the way SMTP connections are processed as a completely separate async task.